roles/pulp_health_check/tasks/main.yml

```diff
@@ -0,0 +1,35 @@
+---
+- meta: flush_handlers
+
+- name: Get services
+  service_facts:
+
+- name: Listing Pulp services
+  debug:
+    msg: '{{ ansible_facts.services|dict2items|selectattr("key", "contains", "pulp")|list|items2dict }}'
+
+
+- name: Checking Pulp services
+  fail:
+    msg: "{{ item.value.name }} state: {{ item.value.state }}"
+  when: "item.value.name != 'pulpcore-worker@.service' and item.value.state != 'running'"
+  with_dict: '{{ ansible_facts.services|dict2items|selectattr("key", "contains", "pulp")|list|items2dict }}'
+
+- name: Ensure Pulp is up and healthy
+  uri:
+    url: "http://{{ pulp_api_bind }}/pulp/api/v3/status/"
+    user: admin
+    password: "{{ pulp_default_admin_password }}"
+    force_basic_auth: yes
+    status_code: "200"
+    validate_certs: "{{ pulp_validate_certs }}"
+  register: result
+  until: >
+    'json' in result and
+    result.json.database_connection.connected is true and
+    result.json.redis_connection.connected is true and
+    result.json.online_workers | map(attribute='name') | select('match', '^resource-manager$') | list | count > 0 and
+    result.json.online_workers | map(attribute='name') | select('match', '^[0-9]+@.*$') | list | count > 0
```

Comment:
I might have said this previously, but I really like the syntax you used, and how you combined this into 1 task (without needing "assert".)

Reply:
I got it from @chouseknecht 
https://github.com/chouseknecht/ansible_galaxy_config/blob/devel/tasks/main.yml

Reply:
Another approach here would be instead of using `and` using a list of checks, something like:

```yaml
  until:
    - 'json' in result
    - result.json.database_connection.connected is true
    - result.json.redis_connection.connected is true
    - result.json.online_workers | map(attribute='name') | select('match', '^resource-manager$') | list | count > 0
    - result.json.online_workers | map(attribute='name') | select('match', '^[0-9]+@.*$') | list | count > 0
```

(grabbed that from https://docs.ansible.com/ansible/latest/user_guide/playbooks_conditionals.html#the-when-statement)